packages/pluggableWidgets/slider-web/src/utils/useOnChangeDebounced.ts

```diff
@@ -0,0 +1,21 @@
+import { Big } from "big.js";
+import { useCallback, useMemo, useRef } from "react";
+import { debounce, executeAction } from "@mendix/piw-utils-internal";
+import { SliderContainerProps } from "../../typings/SliderProps";
+
+export function useOnChangeDebounced(props: SliderContainerProps) {
+    const valueObjRef = useRef(props.valueAttribute);
+    valueObjRef.current = props.valueAttribute;
+
+    const onChangeEnd = useMemo(() => debounce(() => executeAction(props.onChange), 500), [valueObjRef]);
```

Comment:
This seems strange, why would we change the `onChangeEnd` when `valueObjRef` changes? It's also not used inside the memo.

Also: Since you return a fucntion, this can also just be `useCallback`

Reply:
> This seems strange, why would we change the onChangeEnd when valueObjRef changes? It's also not used inside the memo.

Yeah, correct, didn't noticed this dep when copied function.

> Also: Since you return a fucntion, this can also just be useCallback

Well, when I'm wrapping fns with debounce I preffer `useMemo` as in this  case I want to "memo" result of calling debounce

Reply:
> Well, when I'm wrapping fns with debounce I preffer useMemo as in this case I want to "memo" result of calling debounce

Ah yeah I didn't see the `debounce` function, nvm the `useCallback` then.